File: src/main/java/es/mde/tizona/entidades/artefactos/ArtefactoImpl_.java
package es.mde.tizona.entidades.artefactos;

import javax.persistence.metamodel.ListAttribute;
import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

import es.mde.tizona.entidades.artefactos.minas.Mina;

@StaticMetamodel(ArtefactoImpl.class)
public class ArtefactoImpl_ {

	public static volatile SingularAttribute<ArtefactoImpl, String> forma;
	public static volatile SingularAttribute<ArtefactoImpl, String> color;
	public static volatile SingularAttribute<ArtefactoImpl, String> fabricacion;
	public static volatile ListAttribute<ArtefactoImpl, String> encontradoEn;
	public static volatile SingularAttribute<ArtefactoImpl, String> marcasFrio;
	public static volatile SingularAttribute<ArtefactoImpl, String> marcasPintura;
	
	//de mina
	public static volatile SingularAttribute<Mina, String> materialEnvuelta;
}

File: Front-angular/FrontTizona/src/app/fichas/models/artefactos/ArtefactoImpl.ts
import { Artefacto } from "./Artefacto";

export class ArtefactoImpl implements Artefacto {
  id: number;
  nombre: String;
  descripcion: String;
  funcionamiento: String;
  forma: String;
  color: String;
  fabricacion: String;
  encontradoEn: String[];
  listadoImagenes: String[];
  marcasFrio: String;
  marcasPintura: String;

  // constructor(
  //   $nombre: String,
  //   $descripcion: String,
  //   $funcionamiento: String,
  //   $forma: String,
  //   $color: String,
  //   $fabricacion: String,
  //   $encontradoEn: String[],
  //   $listadoImagenes: String[],
  //   $marcasFrio: String,
  //   $marcasPintura: String
  // ) {
  //   this.nombre = $nombre;
  //   this.descripcion = $descripcion;
  //   this.funcionamiento = $funcionamiento;
  //   this.forma = $forma;
  //   this.color = $color;
  //   this.fabricacion = $fabricacion;
  //   this.encontradoEn = $encontradoEn;
  //   this.listadoImagenes = $listadoImagenes;
  //   this.marcasFrio = $marcasFrio;
  //   this.marcasPintura = $marcasPintura;
  // }
}
